sympy/solvers/ode/systems.py

```diff
@@ -0,0 +1,614 @@
+
+    .. math:: r \vec{v} e^{rt} = A \vec{v} e^{rt}
+
+    .. math:: r \vec{v} = A \vec{v}
+
+    where `r` comes out to be eigenvalue of `A` and vector `\vec{v}` is the eigenvector
+    of `A` corresponding to `r`. There are three possibilities of eigenvalues of `A`
+
+    - `n` distinct real eigenvalues
+    - complex conjugate eigenvalues
+    - eigenvalues with multiplicity `k`
+
+    1. When all eigenvalues `r_1,..,r_n` are distinct with `n` different eigenvectors
+    `v_1,...v_n` then the solution is given by
+
+    .. math:: \vec{y} = C_1 e^{r_1 t} \vec{v_1} + C_2 e^{r_2 t} \vec{v_2} +...+ C_n e^{r_n t} \vec{v_n}
+
+    where `C_1,C_2,...,C_n` are arbitrary constants.
+
+    2. When some eigenvalues are complex then in order to make the solution real,
+    we take a linear combination: if `r = a + bi` has an eigenvector
+    `\vec{v} = \vec{w_1} + i \vec{w_2}` then to obtain real-valued solutions to
+    the system, replace the complex-valued solutions `e^{rx} \vec{v}`
+    with real-valued solution `e^{ax} (\vec{w_1} \cos(bx) - \vec{w_2} \sin(bx))`
+    and for `r = a - bi` replace the solution `e^{-r x} \vec{v}` with
+    `e^{ax} (\vec{w_1} \sin(bx) + \vec{w_2} \cos(bx))`
+
+    3. If some eigenvalues are repeated. Then we get fewer than `n` linearly
+    independent eigenvectors, we miss some of the solutions and need to
+    construct the missing ones. We do this via generalized eigenvectors, vectors
+    which are not eigenvectors but are close enough that we can use to write
+    down the remaining solutions. For a eigenvalue `r` with eigenvector `\vec{w}`
+    we obtain `\vec{w_2},...,\vec{w_k}` using
+
+    .. math:: (A - r I) . \vec{w_2} = \vec{w}
+
+    .. math:: (A - r I) . \vec{w_3} = \vec{w_2}
+
+    .. math:: \vdots
+
+    .. math:: (A - r I) . \vec{w_k} = \vec{w_{k-1}}
+
+    Then the solutions to the system for the eigenspace are `e^{rt} [\vec{w}],
+    e^{rt} [t \vec{w} + \vec{w_2}], e^{rt} [\frac{t^2}{2} \vec{w} + t \vec{w_2} + \vec{w_3}],
+    ...,e^{rt} [\frac{t^{k-1}}{(k-1)!} \vec{w} + \frac{t^{k-2}}{(k-2)!} \vec{w_2} +...+ t \vec{w_{k-1}}
+    + \vec{w_k}]`
+
+    So, If `\vec{y_1},...,\vec{y_n}` are `n` solution of obtained from three
+    categories of `A`, then general solution to the system `\vec{y'} = A . \vec{y}`
+
+    .. math:: \vec{y} = C_1 \vec{y_1} + C_2 \vec{y_2} + \cdots + C_n \vec{y_n}
+
+    """
+    eq = match_['eq']
+    func = match_['func']
+    fc = match_['func_coeff']
+    n = len(eq)
+    t = list(list(eq[0].atoms(Derivative))[0].atoms(Symbol))[0]
+    constants = numbered_symbols(prefix='C', cls=Symbol, start=1)
+
+    M = -fc if type(fc) is Matrix else Matrix(n, n, lambda i,j:-fc[i,func[j],0])
+    P, J = matrix_exp_jordan_form(M, t)
+    P = simplify(P)
+    Cvect = Matrix(list(next(constants) for _ in range(n)))
+    sol_vector = P * (J * Cvect)
+
+    sol_vector = [collect(s, ordered(J.atoms(exp)), exact=True) for s in sol_vector]
```

Comment:
We should add a workaround for the collect problem here and a link to the issue (add an explanation and a url to the github issue in a comment). We can do something like:
```julia
In [3]: s = 2*exp(3*t/4) + 3*t*exp(3*t/4)                                                                                         

In [4]: s                                                                                                                         
Out[4]: 
     3⋅t      3⋅t
     ───      ───
      4        4 
3⋅t⋅ℯ    + 2⋅ℯ   

In [5]: exps = s.atoms(exp)                                                                                                       

In [6]: rep = {collect(e, [e]): e for e in exps}                                                                                  

In [7]: collect(s, exps)                                                                                                          
Out[7]: 
              3/4
          ⎛ t⎞   
(3⋅t + 2)⋅⎝ℯ ⎠   

In [8]: collect(s, exps).subs(rep)                                                                                                
Out[8]: 
           3⋅t
           ───
            4 
(3⋅t + 2)⋅ℯ 
```
This problem should not be ignored because it can lead to wrong results.

Reply:
@oscarbenjamin So, the problem here is that the collect function gives us something like:
```exp(t)**Rational(3, 4)``` even when we are doing something like ```collect(exp(3*t/4), [exp(3*t/4])``` whose expected output is ```exp(3*t/4)```. I have to create a new issue regarding this, reference this issue in our new general case solver and the workaround you specified, correct?

Reply:
Looks like this might be fixed soon actually (#19165)